<!--
/************************ LICENCE ***************************
*     This file is part of <ViKM Vital-IT Knowledge Management web application>
*     Copyright (C) <2016> SIB Swiss Institute of Bioinformatics
*
*     This program is free software: you can redistribute it and/or modify
*     it under the terms of the GNU Affero General Public License as
*     published by the Free Software Foundation, either version 3 of the
*     License, or (at your option) any later version.
*
*     This program is distributed in the hope that it will be useful,
*     but WITHOUT ANY WARRANTY; without even the implied warranty of
*     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*     GNU Affero General Public License for more details.
*
*     You should have received a copy of the GNU Affero General Public License
*    along with this program.  If not, see <http://www.gnu.org/licenses/>
*
*****************************************************************/
-->
<div class = 'container'>
<div class="text-left">

  <h2>Documentation</h2>
    MixMHCp2.0 is a webserver to identify motifs within large datasets of peptides.<br>
    The different motifs are mathematically modelled with Position Weight Matrices (PWMs) that are visualized with sequence logos. MixMHCp2.0 includes several features optimized for HLA-I ligands, but it can also be run on peptides from other origins.
    <br>
    <br>
    The algorithm automatically learns the entries of the PWMs and their respective weight.
    The learning is based on mixture models and maximum likelihood (EM algorithm).
    <br>
    <br>
    The command-line script for Mac or Linux can be obtained at:
    <br>
    <a href="https://github.com/GfellerLab/MixMHCp">https://github.com/GfellerLab/MixMHCp</a>.
    <br>

    <h3>Input</h3>
    As input, you should give a list of peptides consisting only of natural amino acids (no gaps).
    Peptides can be of different lengths (from 5 to core_length+10). The algorithm assumes that all peptides of length equal to core length are naturally aligned and that specific positions are found at the beginning (first three positions) and the end (last two positions), as it is the case for HLA-I ligands.
    <br>
    <br>
    The list of peptides can be in standard text format with each peptide on a different line, or in fasta format. Be careful not to load files with metadata, such as Excel spreadsheets.
    <br>
    <br>
    <strong>Max nr of motifs</strong> corresponds to the maximal number of motifs that should be explored. Default value is 6 (as in many HLA peptidomics datasets, although exploring larger number of motifs can often be useful to identify motifs supported by less peptides), maximal value is 50.
    <br>
    <br>
    <strong>Core length</strong> corresponds to the length of the core. The motif deconvolution is first performed on peptides with length equal to core length (these peptides are assumed to be aligned). The motifs inferred by the algorithm (more precisely the first three and last two positions of the motifs) are then used to score peptides of other lengths. Default value is 9.
    <br>
    <br>
    <strong>Trash</strong> integrates a flat motif in the motif deconvolution. Peptides that do not match motifs inferred from the data are assigned to this motif. This is useful to remove potential noise in the data. Default value is Yes.
    <br>
    <br>
    The example provided in input for testing comes from HLA-I peptidomics profiling of HEK293 cells from [Ritz et al. Proteomics 2016]. The optimal number of motifs for this example is equal to 4 (HLA-A03:01 - motifs 1 and 3, HLA-B07:02 - motif 4, HLA-C07:02 - motif 2).
    <br>
    <br>
    If you run MixMHCp2.0 on other pre-aligned peptides than HLA-I ligands, we advise to treat each peptide length separately, setting the core length equal to the peptide length. MixMHCp cannot align the peptides, but we are working on a more general algorithm to identify motifs in non-aligned peptide lists.
    <br>
        
        <h3>Output</h3>
        <h4>Graphical output</h4>
    For each peptide length, the different logos representing the different motif are shown for each total numbers of motifs (from one motif to "Max nr of motifs"), as well as the number of peptides assigned to them. Peptide length distributions are also displayed for each motif.
        
        <h4>Downloadable output</h4>
  <ul>
  <li>The <code>responsibility/</code> folder contains the responsibility of each peptide to each motif for the different number of motifs (resp_n.txt for n motifs).
      These values correspond to the contribution of each peptide to each motif.
      Peptide assignment to each motif can be done by considering the max of the responsibility values.
      <br>
      In addition, for each motif, the optimal position of the beginning and the end of the motif used to score the peptides is shown. This is useful to detect cases of N- or C-terminal extensions. For peptides of length equal to core length, these values are always equal to 1 and "Core length", respectively.
  </li>
      <li>The <code>weights/</code> folder contains information about the weight of each motif for each total number of motifs (<code>weights_n.txt</code>). In addition the peptide length distribution for each motif is shown in <code>weights/plots/</code>. Finally the <code>length_distribution.html</code> shows all peptide length distributions for each motif and for each total number of motifs.
      </li>
      <li>
          The <code>Multiple_PWMs/</code> folder contains the PWMs that optimally model the input data (only of length equal to core length). Of note, these are not meant to be used as predictors, since they do not include any renormalization for background frequencies and the prior used in the mixture model may not be optimal for training predictors.
      </li>
      <li>
          The <code>logos/</code> folder contains the logos (.png) representing each motif for each total number of motifs. Logos have been generated with the <a href="http://baderlab.org/Software/LOLA">LoLa</a> software. The <code>logos.html</code> file allows you to quickly visualize all the logos of length equal to core length. Logos for other peptide lengths can be seen in <code>logos_html/</code> folder.
      </li>
        <li>The <code>data/</code> folder contains the input data. </li>
        <li>
            The <code>KLD/best_ncl.txt</code> file contains the KLD scores for each number of motifs [Andreatta et al. Bioinformatics 2013], as well as the predicted optimal number of motifs [see Bassani-Sternberg and Gfeller, J Immunol. 2016]. However, it is in general advised to manually go through the motifs to check the optimal number of motifs.
        </li>
      </ul>
      
        <h2>Other information</h2>
        MixMHCp algorithm is developed by the <a href="http://www.unil.ch/dof/research/gfeller">Computational Cancer Biology Lab</a> at the Ludwig Center for Cancer Research at <a href="http://unil.ch/index.html">UNIL</a> led by Prof. David Gfeller. The website is developed in collaboration with the Vital-IT team.
    <br>
      <br>
    In the past, the method has been successfully applied to phage display data to identify multiple specificity patterns [Gfeller et al, MSB 2011; Kim et al. NAR 2012] and HLA-I peptidomics data to deconvolute mixtures of HLA-I ligands coming from up to 6 alleles [Bassani-Sternberg and Gfeller, JI 2016; Bassani-Sternberg et al. PLoS Comp Bio 2017; Guillaume et al. PNAS 2018, Gfeller et al. BioRxiv 2018].
    <br>
        <br>
    If you use this website in a publication, please cite <a href="https://www.ncbi.nlm.nih.gov/pubmed/27511729">[Bassani-Sternberg M and Gfeller D, J. Immunol. 197(6):2492 (2016)]</a> and [Gfeller et al. BioRxiv 2018].
        <br>
        <br>

    MixMHCp can be used freely by academic groups for non-commercial purposes. The product is provided free of charge, and, therefore, on an "as is" basis, without warranty of any kind.
    <br>
    If you plan to use MixMHCp in any for-profit application, you are required to obtain a separate license. To do so, please contact eauffarth@licr.org or lfoit@licr.org at the Ludwig Institute for Cancer Research Ltd.
    <br>


	</div>




</div>
